google/cloud/aiplatform/featurestore/featurestore.py

```diff
@@ -1210,34 +1216,54 @@ def batch_serve_to_df(
 
         self.wait()
         featurestore_name_components = self._parse_resource_name(self.resource_name)
-        featurestore_id = featurestore_name_components["featurestore"]
-
-        temp_bq_dataset_name = f"temp_{featurestore_id}_{uuid.uuid4()}".replace(
-            "-", "_"
-        )
 
-        project_id = resource_manager_utils.get_project_id(
-            project_number=featurestore_name_components["project"],
-            credentials=self.credentials,
-        )
-        temp_bq_dataset_id = f"{project_id}.{temp_bq_dataset_name}"[:1024]
-        temp_bq_dataset = bigquery.Dataset(dataset_ref=temp_bq_dataset_id)
-        temp_bq_dataset.location = self.location
-        temp_bq_dataset = bigquery_client.create_dataset(temp_bq_dataset)
+        # if user didn't specify BigQuery dataset, create an ephemeral one
+        if bq_dataset_id is None:
+            temp_bq_full_dataset_id = self._get_ephemeral_bq_full_dataset_id(
+                featurestore_name_components["featurestore"],
+                featurestore_name_components["project"],
+            )
+            temp_bq_dataset = self._create_ephemeral_bq_dataset(
+                bigquery_client, temp_bq_full_dataset_id
+            )
+            temp_bq_batch_serve_table_name = "batch_serve"
+            temp_bq_read_instances_table_name = "read_instances"
+
+        # if user specified BigQuery dataset, create ephemeral tables
+        else:
+            temp_bq_full_dataset_id = bq_dataset_id
+            temp_bq_dataset = bigquery.Dataset(dataset_ref=temp_bq_full_dataset_id)
+            temp_bq_batch_serve_table_name = f"tmp_batch_serve_{uuid.uuid4()}".replace(
+                "-", "_"
+            )
+            temp_bq_read_instances_table_name = (
+                f"tmp_read_instances_{uuid.uuid4()}".replace("-", "_")
+            )
 
-        temp_bq_batch_serve_table_name = "batch_serve"
-        temp_bq_read_instances_table_name = "read_instances"
         temp_bq_batch_serve_table_id = (
-            f"{temp_bq_dataset_id}.{temp_bq_batch_serve_table_name}"
+            f"{temp_bq_full_dataset_id}.{temp_bq_batch_serve_table_name}"
         )
+
         temp_bq_read_instances_table_id = (
-            f"{temp_bq_dataset_id}.{temp_bq_read_instances_table_name}"
+            f"{temp_bq_full_dataset_id}.{temp_bq_read_instances_table_name}"
         )
 
         try:
 
+            # partial config to ensure the timestamp column has Timestamp and
+            #   not Datetime
+            job_config = bigquery.LoadJobConfig(
```

Comment:
This seems out-of-scope for the proposed change. Why was this included?

Reply:
busted; I'll pull that out and create separate issue/pr :)

Reply:
removed this in [a8f052e](https://github.com/googleapis/python-aiplatform/pull/1623/commits/a8f052e734d0b68fc8d85e852df13a2f1417078f)